hawkbit-rest/hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DataConversionHelper.java

```diff
@@ -104,7 +104,7 @@ private static DdiArtifact createArtifact(final Target target, final ArtifactUrl
                         new SoftwareData(artifact.getSoftwareModule().getId(), artifact.getFilename(), artifact.getId(),
                                 artifact.getSha1Hash())),
                         ApiType.DDI, request.getURI())
-                .forEach(entry -> file.add(new Link(entry.getRef()).withRel(entry.getRel())));
+                .forEach(entry -> file.add(Link.of(entry.getRef()).withRel(entry.getRel())));
```

Comment:
Pretty sure this is causing the issue i saw in https://github.com/eclipse/hawkbit/pull/1320#issuecomment-1445236485

This needs an `.expand()` call at the end to drop the optional templated arguments.